src/backend/access/transam/xlog.c

```diff
@@ -5945,6 +5945,13 @@ pauseRecoveryOnRestorePoint(XLogReaderState *record)
 			SetRecoveryPause(true);
 			recoveryPausesHere();
 
+			/*
+			* GPDB: When hot_standby is 'on', to avoid getting
+			* stuck during a possible timeline conflict,
+			* rescan for new timelines and update recovery target TLI.
+			*/
+			rescanLatestTimeLine();
```

Comment:
A rescan will update the global target TLI which is not ideal for `recovery_target_timeline` set to "current" or a numeric value.  We should probably add a conditional here to restrict this logic to `recovery_target_timeline=latest`.  Recommendation:
```suggestion
			/*
			 * The gp_pause_on_restore_point_replay GUC may have been updated
			 * to a new restore point that exists on a timeline that we aren't
			 * aware of yet. Check the off-line archival storage for any new
			 * timeline history files.
			 */
			if (recoveryTargetTimeLineGoal == RECOVERY_TARGET_TIMELINE_LATEST)
				rescanLatestTimeLine();
```

Also, this logic should probably go under the below promotion logic and a `return;` should be added to the promotion logic.  If we get a promotion request, we probably don't care about checking for a new timeline.